src/current/v22.1/bulk-delete-data.md

```diff
@@ -76,22 +98,185 @@ while True:
+                ps.setInt(1, warehouseId);
+                ResultSet results = ps.executeQuery();
+                if (!results.next()) {
+                    cont = false;
+                } else {
+                    results.last();
+                    System.out.println("Deleted " + results.getRow() + " rows.");
+                    // get the warehouse ID of the last row of this batch
+                    warehouseId = results.getInt(1);
+                    System.out.println("Warehouse ID is now " + warehouseId);
+                }
+            } while (cont);
+        }
+        catch(Exception e) {
+            return;
+        }
+    }
+
+    public static void main(String[] args) throws SQLException {
+        Connection connection = null;
+        try {
+            // create the datasource for the JDBC driver
+            PGSimpleDataSource ds = new PGSimpleDataSource();
+            ds.setApplicationName("docs_bulk_delete_java");
+            // get the cluster JDBC URL from an environment variable
+            ds.setUrl(Optional.ofNullable(System.getenv("JDBC_DATABASE_URL")).orElseThrow(
+                  () -> new IllegalArgumentException("JDBC_DATABASE_URL is not set.")));
+            connection = ds.getConnection();
+            // call the method to perform the deletes
+            deleteData(connection);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            // close the connection
+            if (connection != null) connection.close();
+        }
+    }
+}
+~~~
+
+</section>
+
+<section class="filter-content" markdown="1" data-scope="csharp">
+
+A simple C# Npgsql application that bulk deletes rows in batches of 5000 would look like this:
+
+{% include_cached copy-clipboard.html %}
+~~~ csharp
+using System;
+using System.Data;
+using System.Net.Security;
+using Npgsql;
+
+namespace Cockroach
+{
+  class MainClass
+  {
+    static void Main(string[] args)
+    {
+      // create the connection string from the connection parameters
+      var connStringBuilder = new NpgsqlConnectionStringBuilder();
+      connStringBuilder.Host = "cluster-name-743.g95.cockroachlabs.cloud";
+      connStringBuilder.Port = 26257;
+      connStringBuilder.SslMode = SslMode.VerifyFull;
+      connStringBuilder.Username = "maxroach";
+      connStringBuilder.Password = "notAGoodPassword";
+      connStringBuilder.Database = "tpcc";
+      connStringBuilder.ApplicationName = "docs_bulk_delete_csharp";
+      // call the method to perform the deletes
+      DeleteRows(connStringBuilder.ConnectionString);
+    }
+
+    static void DeleteRows(string connString)
+    {
+      // create the data source with the connection string
+      using var dataSource = NpgsqlDataSource.Create(connString);
+      {
+        using var connection = dataSource.OpenConnection();
+        bool cont = true;
```

Comment:
Use `var` everywhere for consistency (unless this is invalid here - I have not written c# in a long time).

Reply:
`var` is used in lieu of the type, so I think it would be cleaner to keep `bool`.